lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -390,7 +390,8 @@ export abstract class CacheManager implements ICacheManager {
                 return;
             }
 
-            if (!!homeAccountId && !this.matchHomeAccountId(entity, homeAccountId)) {
+            // homeAccountId can be an empty string, and !!("") = false
```

Comment:
Consider adding a comment more about the WHY and not the WHAT, i.e. 

"In client_credentials, homeAccountId is null, and the logic should filter out tokens with homeAccountId, because these are user tokens"